homeassistant/components/withings/binary_sensor.py

```diff
@@ -24,9 +26,20 @@ async def async_setup_entry(
         entry.entry_id
     ][BED_PRESENCE_COORDINATOR]
 
-    entities = [WithingsBinarySensor(coordinator)]
+    ent_reg = er.async_get(hass)
 
-    async_add_entities(entities)
+    def _async_add_bed_presence_entity() -> None:
+        """Add bed presence entity."""
+        entities = [WithingsBinarySensor(coordinator)]
+
+        async_add_entities(entities)
+
+    if ent_reg.async_get_entity_id(
+        Platform.BINARY_SENSOR, DOMAIN, f"withings_{entry.unique_id}_in_bed"
+    ):
+        _async_add_bed_presence_entity()
+    else:
+        coordinator.async_add_listener(_async_add_bed_presence_entity)
```

Comment:
One problem I am experiencing with this setup is that this is fired at every update, thus creating more entities. I can't really find a nice way to remove this listener after we added the sensor

Reply:
Gave it a shot, curious if this is what we want